src/functions.php

```diff
@@ -344,3 +346,44 @@ function _current_time()
 {
     return function_exists('hrtime') ? hrtime(true) / 1e9 : microtime(true);
 }
+
+
+/**
+ * @param int $options
+ * @return UriInterface
+ * @internal
+ */
+function _idn_uri_convert(UriInterface $uri, $options = IDNA_DEFAULT)
+{
+    if ($uri->getHost()) {
+        $asciiHost = idn_to_ascii($uri->getHost(), $options, INTL_IDNA_VARIANT_UTS46, $info);
```

Comment:
It make sense to use `$variant = defined('INTL_IDNA_VARIANT_UTS46') ? INTL_IDNA_VARIANT_UTS46 : null;` workaround here, because it seems that the constant is not always available.

Reply:
Done.

Reply:
Done.